src/realmModels/ObservationPhoto.js

```diff
@@ -32,7 +32,9 @@ class ObservationPhoto extends Realm.Object {
 
   static mapPhotoForUpload( observtionID, observationPhoto ) {
     return {
-      "photo[uuid]": observationPhoto.uuid,
+      photo: {
+        id: observationPhoto.photo.id
+      },
```

Comment:
Sending the ObservationPhoto UUID instead of the Photo UUID is dangerous here, because they will be different in server responses. The server doesn't seem to respond with the photo UUID at all, which is a separate problem.